Tools/clinic/clinic.py

```diff
@@ -4685,9 +4685,6 @@ def state_modulename_name(self, line: str) -> None:
         # this line is permitted to start with whitespace.
         # we'll call this number of spaces F (for "function").
 
-        if not self.valid_line(line):
-            return
```

Comment:
This line was unreachable: `state_modulename_name()` was only ever called from `state_dsl_start`, but `state_dsl_start` never gets to the bit where it calls `state_modulename_name` if it's not a valid line:

https://github.com/python/cpython/blob/407d7fda94268d70b92e2e685ca9891ec3641e78/Tools/clinic/clinic.py#L4654-L4656

Once it's determined that it _is_ a valid line, the line is just passed along to `state_modulename_name` without it being altered at all, so there's no possible way it could be an invalid line at this point